app/models/spree/payment.rb

```diff
@@ -176,8 +184,14 @@ def create_payment_profile
       gateway_error e
     end
 
+    # Makes newly entered payments invalidate previously entered payments so the most recent payment
+    # is used by the gateway.
     def invalidate_old_payments
-      order.payments.with_state('checkout').where("id != ?", id).each(&:invalidate!)
+      order.payments.with_state('checkout').where("id != ?", id).each do |payment|
```

Comment:
I think we can start using `#not` in Rails 4 for these "where not equals" AR scopes. Not blocking, just mentioning it...

Reply:
wooow! I totally forgot Rails 4.0 introduced [#not](https://apidock.com/rails/v4.0.2/ActiveRecord/QueryMethods/WhereChain/not)!